src/common/random-bytes-browser.js

```diff
@@ -1,4 +1,15 @@
-const Crypto = window.crypto;
+function resolveCrypto() {
+	if(typeof window !== 'undefined' && window.crypto != null) {
+		// UI thread
+		return window.crypto;
+	} else if (typeof self !== 'undefined' && self.crypto != null) {
+		// Worker thread
+		return self.crypto;
+	} else {
+		throw new Error('crypto not found');
+	}
+}
+const Crypto = resolveCrypto();
```

Comment:
good catch.  it looks like we can reduce this to:
```javascript
const Crypto =
	try { window.crypto } ||
	try { self.crypto } ||
		function () { throw new Error('crypto not supported') }()
```
even with that there's still a potential bug where `getRandomValues` is not a function, but that i'm not too worried

Reply:
looks like js is more finicky than anticipated:
```js
const Crypto =
	(function () { try { return window.crypto } catch {} }){} ||
	(function () { try { return window.crypto } catch {} }){} ||
		function () { throw new Error('crypto not supported') }()
```
could probably make 2 functions to help here:
- safeExpression: take a function and execute it in an exception-safe context. possibly take a default to return in the catch
- throwError: takes an error and throws it.  may benefit from a little stack rewriting so that it acts transparently.

Also, if you want to check if something is undefined, use `void(null) === foo`.  `void` is reserved so it can't be redefined, even as a function variable.  then you can use natural equivalence rather than string comparison

Reply:
Good feedback, in the end I went for a different approach: using `self` seems the way to go to resolve correctly in window/worker.

More info (well supported):
-  https://developer.mozilla.org/en-US/docs/Web/API/Window/self#browser_compatibility
-  https://caniuse.com/?search=self

PR updated.